registry/extension/oras/referrers.go

```diff
@@ -2,25 +2,53 @@ package oras
 
 import (
 	"encoding/json"
+	"fmt"
 	"net/http"
+	"strconv"
+	"strings"
 
 	"github.com/distribution/distribution/v3"
 	dcontext "github.com/distribution/distribution/v3/context"
 	"github.com/distribution/distribution/v3/registry/api/errcode"
 	v2 "github.com/distribution/distribution/v3/registry/api/v2"
+	"github.com/opencontainers/go-digest"
 	orasartifacts "github.com/oras-project/artifacts-spec/specs-go/v1"
 )
 
-// referrersResponse describes the response body of the referrers API.
-type referrersResponse struct {
-	Referrers []orasartifacts.Descriptor `json:"references"`
+// ReferrersResponse describes the response body of the referrers API.
+type ReferrersResponse struct {
+	Referrers []orasartifacts.Descriptor `json:"referrers"`
 }
 
+const maxPageSize = 50
+
+// minimum page size used for # of digests to put in nextToken
+const minPageSize = 3
+
 func (h *referrersHandler) getReferrers(w http.ResponseWriter, r *http.Request) {
 	dcontext.GetLogger(h.extContext).Debug("Get")
 
 	// This can be empty
 	artifactType := r.FormValue("artifactType")
+	nPage, err := strconv.Atoi(r.FormValue("n"))
+	// client specified nPage must be greater than min page size
+	if nPage < minPageSize || err != nil {
```

Comment:
`|| nPage > maxPageSize`?

Reply:
updated to use `maxPageSize` if `n > maxPageSize`